image building is broken re verifying go versions

See e.g. https://github.com/kcp-dev/kcp/runs/7865988023?check_suite_focus=true:

```
2022-08-16T19:54:10.5122613Z [1/2] STEP 17/17: RUN CGO_ENABLED=0 make
2022-08-16T19:54:10.6802261Z time="2022-08-16T19:54:10Z" level=warning msg="Failed to decode the keys [\"machine\"] from \"/usr/share/containers/containers.conf\"."
2022-08-16T19:54:10.8581828Z hack/verify-go-versions.sh
2022-08-16T19:54:10.8679016Z grep: .ci-operator.yaml: No such file or directory
2022-08-16T19:54:10.8688060Z Wrong go version in .ci-operator.yaml, expected 1.18
2022-08-16T19:54:10.8688747Z make: *** [Makefile:254: verify-go-versions] Error 1
2022-08-16T19:54:12.4770598Z error building at STEP "RUN CGO_ENABLED=0 make": error while running runtime: exit status 2
```

/kind bug
/assign

--- a/Dockerfile
+++ b/Dockerfile
@@ -35,6 +35,10 @@ COPY cmd/ cmd/
 COPY third_party/ third_party/
 COPY hack/ hack/
 COPY .git/ .git/
+# To make sure hack/verify-go-versions.sh succeeds
+COPY .ci-operator.yaml .
+COPY Dockerfile .
+COPY .github/ .github/
 
 RUN apt-get update && apt-get install -y jq && mkdir bin
 RUN CGO_ENABLED=0 make

--- a/Makefile
+++ b/Makefile
@@ -249,7 +249,7 @@ verify-k8s-deps:
 verify-imports:
 	hack/verify-imports.sh
 
-.PHONY: verify-go-imports
+.PHONY: verify-go-versions
 verify-go-versions:
 	hack/verify-go-versions.sh
 

